### PR TITLE
OBAM-241 Validate shipment against quantity available to promise

### PR DESCRIPTION
**Pull request by awalkowiak**

### CODE REVIEW

**Comment by awalkowiak** [comment]
@jmiranda during review, please take a look at my comment in the OBAM-241 ticket.

**Comment by jmiranda** [comment]
@awalkowiak is this one ready? it still has the `donotmerge` label.

File: grails-app/services/org/pih/warehouse/shipping/ShipmentService.groovy [hunk 1]
```diff
@@ -704,16 +703,22 @@ class ShipmentService {
                 throw new ValidationException("Shipment item is invalid", shipmentItem.errors)
             }
 
-            // Check whether there's any stock in the bin location for the given inventory item
-            def quantityOnHand = getQuantityOnHand(origin, shipmentItem.binLocation, shipmentItem.inventoryItem, binLocationRequired)
+            def quantityAvailableToPromise = productAvailabilityService.getQuantityAvailableToPromise(origin, shipmentItem.binLocation, shipmentItem.inventoryItem)
+            def quantityPicked
+            if (shipmentItem?.binLocation) {
```
Comment (jmiranda) [inline grails-app/services/org/pih/warehouse/shipping/ShipmentService.groovy:708]: Is it possible to clean this up? Maybe move it to the shipmentItem
```
def quantityPicked = shipmentItem.quantityPicked
```

Comment (jmiranda) [reply grails-app/services/org/pih/warehouse/shipping/ShipmentService.groovy:708]: where quantityPicked is a getter method getQuantityPicked that'll look something like this
```
def getQuantityPicked() { 
    if (binLocation) { 
        return requisitionItem?.picklistItems?.findAll { it.inventoryItem == shipmentItem?.inventoryItem && it.binLocation == shipmentItem?.binLocation }?.sum { it.quantity }
    } else {
        return shipmentItem?.requisitionItem?.picklistItems?.findAll { it.inventoryItem == shipmentItem?.inventoryItem }?.sum { it.quantity }
    }
}
```

**Review by jmiranda — APPROVED** [review]
LGTM

**Review by pmuchowski — APPROVED** [review]

**Pull request merged**